segment toggle functionality doesn't work in 24.0.4

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

Since i upgraded from  23.20.3 -> 24.0.4 the segment toggle functionality doesn't work 
` oh-my-posh toggle spotify` no longer toggles the spotify segment also `oh-my-posh toggle kubectl` doesnt toggle spotify etc 

### Theme

mojada

### What OS are you seeing the problem on?

macOS

### Which shell are you using?

powershell

### Log output

```Shell
oh-my-posh debug pwsh --plain                                               66ms rancher-desktop 󰏤 Social Distortion - Ball and Chain  86%  09:09 (Tue) 

Version: 24.0.4

Shell: pwsh (7.4.6)

Prompt:


 pwsh   the-mentor   ~  $                                                                              0ms rancher-desktop 󰏤 Social Distortion - Ball and Chain  85%  09:10 (Tue) 

Segments:

ConsoleTitle(true)                         -   0 ms
Shell(true)                                -   0 ms
Os(true)                                   -   0 ms
Session(true)                              -   5 ms
Root(false)                                -   0 ms
Path(true)                                 -   0 ms
Git(false)                                 -   0 ms
Text(true)                                 -   0 ms
Status(true)                               -   0 ms
Executiontime(true)                        -   0 ms
Kubectl(true)                              -   1 ms
Terraform(false)                           -   1 ms
Spotify(true)                              - 372 ms
Battery(true)                              -  12 ms
Time(true)                                 -   0 ms

Run duration: 398.485917ms

Cache path: /Users/the-mentor/.cache/oh-my-posh

Config path: /Users/the-mentor/mojada.omp.json

Logs:

[DEBUG] 09:10:03.124 debug.go:50 → debug mode enabled
[DEBUG] 09:10:03.125 debug.go:50 → plain mode enabled
[DEBUG] 09:10:03.125 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.125 terminal.go:Getenv(LOCALAPPDATA) - 561.875µs
[DEBUG] 09:10:03.125 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.125 terminal.go:Getenv(OMP_CACHE_DIR) - 1.791µs
[DEBUG] 09:10:03.125 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.125 terminal.go:Getenv(XDG_CACHE_HOME) - 1.25µs
[TRACE] 09:10:03.125 terminal.go:CachePath() - 586.459µs
[DEBUG] 09:10:03.125 terminal.go:65 → loading cache file: /Users/the-mentor/.cache/oh-my-posh/omp.cache
[DEBUG] 09:10:03.125 terminal.go:65 → loading cache key: environment_platform
[DEBUG] 09:10:03.125 terminal.go:65 → loading cache key: is_wsl
[TRACE] 09:10:03.125 terminal.go(/Users/the-mentor/.cache/oh-my-posh/omp.cache) - 98.917µs
[DEBUG] 09:10:03.125 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.125 terminal.go:Getenv(LOCALAPPDATA) - 1.542µs
[DEBUG] 09:10:03.125 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.125 terminal.go:Getenv(OMP_CACHE_DIR) - 1.25µs
[DEBUG] 09:10:03.125 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.125 terminal.go:Getenv(XDG_CACHE_HOME) - 1.375µs
[TRACE] 09:10:03.125 terminal.go:CachePath() - 15.834µs
[DEBUG] 09:10:03.125 terminal.go:65 → loading cache file: /Users/the-mentor/.cache/oh-my-posh/omp.cache.39fa3415-6ec4-49c6-9c5f-49a732414d95
[TRACE] 09:10:03.126 terminal.go(/Users/the-mentor/.cache/oh-my-posh/omp.cache.39fa3415-6ec4-49c6-9c5f-49a732414d95) - 37.792µs
[TRACE] 09:10:03.126 terminal.go:setPromptCount() - 250ns
[TRACE] 09:10:03.126 terminal.go:GOOS() - 42ns
[DEBUG] 09:10:03.126 terminal.go:setPwd:194 → /Users/the-mentor
[TRACE] 09:10:03.126 terminal.go:setPwd() - 7.5µs
[DEBUG] 09:10:03.126 terminal.go:Getenv:161 → /Users/the-mentor/mojada.omp.json
[TRACE] 09:10:03.126 terminal.go:Getenv(POSH_THEME) - 1.417µs
[DEBUG] 09:10:03.126 terminal.go:ResolveConfigPath:93 → config set using POSH_THEME: /Users/the-mentor/mojada.omp.json
[TRACE] 09:10:03.126 terminal.go:ResolveConfigPath() - 5.834µs
[TRACE] 09:10:03.126 terminal.go:Init() - 1.908042ms
[TRACE] 09:10:03.126 terminal.go:Shell() - 208ns
[TRACE] 09:10:03.126 terminal.go:Flags() - 0s
[TRACE] 09:10:03.126 load.go:loadConfig() - 479.209µs
[TRACE] 09:10:03.126 terminal.go:Flags() - 42ns
[DEBUG] 09:10:03.126 debug.go:60 → terminal program: iTerm.app
[DEBUG] 09:10:03.126 debug.go:60 → terminal shell: shell
[DEBUG] 09:10:03.126 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.126 terminal.go:Getenv(OMP_CACHE_DISABLED) - 1.208µs
[DEBUG] 09:10:03.135 terminal.go:RunCommand:371 → 4
[TRACE] 09:10:03.135 terminal.go:RunCommand(defaults read -g AppleAccentColor) - 8.452917ms
[TRACE] 09:10:03.135 colors.go:SetAccentColor() - 8.459583ms
[TRACE] 09:10:03.135 terminal.go:Shell() - 83ns
[DEBUG] 09:10:03.135 terminal.go:Getenv:161 → 7.4.6
[TRACE] 09:10:03.135 terminal.go:Getenv(POSH_SHELL_VERSION) - 4.75µs
[DEBUG] 09:10:03.135 debug.go:PrintDebug:23 → segment: Title
[TRACE] 09:10:03.135 terminal_unix.go:Root() - 375ns
[TRACE] 09:10:03.135 terminal.go:Shell() - 42ns
[TRACE] 09:10:03.135 terminal.go:StatusCodes() - 208ns
[DEBUG] 09:10:03.136 terminal_unix.go:IsWsl:35 → false
[TRACE] 09:10:03.136 terminal_unix.go:IsWsl() - 4.792µs
[DEBUG] 09:10:03.136 terminal_unix.go:IsWsl:35 → false
[TRACE] 09:10:03.136 terminal_unix.go:IsWsl() - 1.5µs
[TRACE] 09:10:03.136 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.136 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.136 terminal.go:GOOS() - 42ns
[TRACE] 09:10:03.136 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.136 terminal.go:GOOS() - 41ns
[TRACE] 09:10:03.136 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.136 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.136 terminal.go:GOOS() - 41ns
[TRACE] 09:10:03.136 terminal.go:GOOS() - 41ns
[TRACE] 09:10:03.136 terminal.go:GOOS() - 41ns
[TRACE] 09:10:03.136 terminal.go:GOOS() - 42ns
[DEBUG] 09:10:03.136 terminal.go:User:331 → the-mentor
[TRACE] 09:10:03.136 terminal.go:User() - 1.875µs
[DEBUG] 09:10:03.136 terminal.go:Host:348 → DKMT7NKHRP.corp.proofpoint.com
[TRACE] 09:10:03.136 terminal.go:Host() - 7.875µs
[TRACE] 09:10:03.136 terminal.go:GOOS() - 0s
[DEBUG] 09:10:03.136 terminal.go:Getenv:161 → 1
[TRACE] 09:10:03.136 terminal.go:Getenv(SHLVL) - 1.458µs
[TRACE] 09:10:03.136 terminal.go:TemplateCache() - 214.041µs
[TRACE] 09:10:03.136 text.go:Render({{.UserName}}@{{.HostName}} : {{.Folder}}) - 375.625µs
[TRACE] 09:10:03.136 terminal.go:Flags() - 41ns
[DEBUG] 09:10:03.136 terminal.go:Getenv:161 → 26
[TRACE] 09:10:03.136 terminal.go:Getenv(POSH_CURSOR_LINE) - 150.291µs
[DEBUG] 09:10:03.136 terminal.go:Getenv:161 → 1
[TRACE] 09:10:03.136 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 1.75µs
[TRACE] 09:10:03.136 terminal.go:Flags() - 0s
[TRACE] 09:10:03.136 terminal.go:Flags() - 41ns
[TRACE] 09:10:03.136 terminal.go:Flags() - 42ns
[TRACE] 09:10:03.136 terminal.go:GOOS() - 125ns
[DEBUG] 09:10:03.136 segment.go:Execute:103 → segment: Text
[TRACE] 09:10:03.136 terminal.go:TemplateCache() - 83ns
[TRACE] 09:10:03.136 terminal.go:Flags() - 83ns
[TRACE] 09:10:03.136 terminal.go:GOOS() - 42ns
[DEBUG] 09:10:03.136 segment.go:Execute:103 → segment: Shell
[DEBUG] 09:10:03.136 shell.go:Enabled:26 → mapped_shell_names: map[]
[TRACE] 09:10:03.136 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.136 terminal.go:GOOS() - 0s
[DEBUG] 09:10:03.136 segment.go:Execute:103 → segment: Path
[DEBUG] 09:10:03.136 segment.go:Execute:103 → segment: Git
[TRACE] 09:10:03.136 terminal.go:Flags() - 42ns
[TRACE] 09:10:03.136 terminal.go:Shell() - 0s
[DEBUG] 09:10:03.136 terminal_unix.go:IsWsl:35 → false
[TRACE] 09:10:03.136 terminal.go:Flags() - 0s
[DEBUG] 09:10:03.136 path.go:145 → display_cygpath: false
[TRACE] 09:10:03.136 terminal.go:TemplateCache() - 42ns
[TRACE] 09:10:03.136 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.136 terminal_unix.go:IsWsl() - 2.125µs
[TRACE] 09:10:03.136 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.136 terminal.go:GOOS() - 84ns
[TRACE] 09:10:03.136 terminal_unix.go:IsWsl2() - 6.667µs
[TRACE] 09:10:03.136 terminal.go:PathSeparator() - 1.375µs
[TRACE] 09:10:03.136 terminal.go:GOOS() - 42ns
[TRACE] 09:10:03.136 terminal.go:Shell() - 0s
[DEBUG] 09:10:03.136 segment.go:Execute:103 → segment: Session
[TRACE] 09:10:03.136 terminal.go:Flags() - 0s
[TRACE] 09:10:03.136 terminal.go:GOOS() - 41ns
[DEBUG] 09:10:03.136 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.136 terminal.go:Getenv(SSH_CONNECTION) - 2.542µs
[DEBUG] 09:10:03.136 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.136 terminal.go:Getenv(SSH_CLIENT) - 1.292µs
[DEBUG] 09:10:03.136 path.go:setMappedLocations:552 → mapped_locations_enabled: true
[DEBUG] 09:10:03.136 path.go:setMappedLocations:553 → 
[DEBUG] 09:10:03.136 path.go:setMappedLocations:554 → 
[TRACE] 09:10:03.136 terminal.go:GOOS() - 41ns
[TRACE] 09:10:03.136 terminal.go:PathSeparator() - 1.417µs
[TRACE] 09:10:03.136 terminal.go:GOOS() - 41ns
[DEBUG] 09:10:03.137 terminal.go:CommandPath:395 → /usr/bin/git
[TRACE] 09:10:03.137 terminal.go:CommandPath(git) - 136.208µs
[TRACE] 09:10:03.137 terminal.go:HasCommand(git) - 138µs
[DEBUG] 09:10:03.137 terminal_unix.go:Platform:95 → darwin
[DEBUG] 09:10:03.137 git.go:shouldDisplay:326 → fetch_bare_info: false
[DEBUG] 09:10:03.137 terminal.go:ResolveSymlink:287 → /Users/the-mentor
[TRACE] 09:10:03.137 terminal.go:ResolveSymlink(/Users/the-mentor) - 13.666µs
[ERROR] 09:10:03.137 terminal.go:HasParentFilePath:559 → stat .git: no such file or directory
[TRACE] 09:10:03.137 terminal.go:HasParentFilePath(.git) - 33.041µs
[TRACE] 09:10:03.137 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.137 terminal.go:GOOS() - 0s
[DEBUG] 09:10:03.137 path.go:setMappedLocations:555 → ~
[DEBUG] 09:10:03.137 path.go:setMappedLocations:560 → mapped_locations: map[]
[TRACE] 09:10:03.137 terminal.go:GOOS() - 41ns
[TRACE] 09:10:03.137 terminal.go:PathSeparator() - 1.667µs
[TRACE] 09:10:03.137 terminal.go:GOOS() - 42ns
[TRACE] 09:10:03.137 terminal.go:GOOS() - 42ns
[TRACE] 09:10:03.137 terminal.go:GOOS() - 41ns
[TRACE] 09:10:03.137 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.137 terminal.go:PathSeparator() - 1.208µs
[TRACE] 09:10:03.137 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.137 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.137 terminal.go:GOOS() - 41ns
[TRACE] 09:10:03.137 terminal.go:GOOS() - 42ns
[TRACE] 09:10:03.137 terminal.go:GOOS() - 41ns
[TRACE] 09:10:03.137 terminal.go:GOOS() - 42ns
[TRACE] 09:10:03.137 terminal.go:GOOS() - 42ns
[TRACE] 09:10:03.137 terminal.go:GOOS() - 42ns
[TRACE] 09:10:03.137 terminal.go:Flags() - 41ns
[DEBUG] 09:10:03.137 path.go:colorizePath:728 → cycle: []
[DEBUG] 09:10:03.137 path.go:getFolderSeparator:263 → NO DATA
[DEBUG] 09:10:03.137 path.go:getFolderSeparator:265 → /
[DEBUG] 09:10:03.137 path.go:colorizePath:731 → cycle_folder_separator: false
[DEBUG] 09:10:03.137 path.go:colorizePath:732 → %s
[DEBUG] 09:10:03.137 path.go:colorizePath:734 → %s
[DEBUG] 09:10:03.137 path.go:colorizePath:735 → %s
[DEBUG] 09:10:03.137 path.go:colorizePath:736 → %s
[TRACE] 09:10:03.137 terminal.go:TemplateCache() - 0s
[TRACE] 09:10:03.137 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.137 terminal.go:StackCount() - 83ns
[TRACE] 09:10:03.137 terminal_unix.go:DirIsWritable(/Users/the-mentor) - 12.375µs
[TRACE] 09:10:03.137 terminal.go:TemplateCache() - 42ns
[TRACE] 09:10:03.137 terminal.go:TemplateCache() - 42ns
[TRACE] 09:10:03.137 text.go:Render( {{ .Name }} ) - 884.084µs
[TRACE] 09:10:03.137 text.go:Render(diamond) - 125ns
[TRACE] 09:10:03.137 text.go:Render( {{ if .WSL }}WSL at {{ end }}{{.Icon}} ) - 15.292µs
[TRACE] 09:10:03.137 text.go:Render(diamond) - 42ns
[DEBUG] 09:10:03.142 terminal.go:RunCommand:371 → the-mentor                     Nov  5 09:10
[TRACE] 09:10:03.142 terminal.go:RunCommand(who am i) - 5.71725ms
[TRACE] 09:10:03.142 terminal.go:TemplateCache() - 0s
[TRACE] 09:10:03.142 terminal.go:TemplateCache() - 0s
[TRACE] 09:10:03.142 text.go:Render({{ .UserName }}</> ) - 125.125µs
[TRACE] 09:10:03.142 text.go:Render(powerline) - 83ns
[TRACE] 09:10:03.142 text.go:Render(powerline) - 84ns
[TRACE] 09:10:03.142 text.go:Render(  {{ .Path }} ) - 13.375µs
[TRACE] 09:10:03.142 text.go:Render(powerline) - 42ns
[TRACE] 09:10:03.142 text.go:Render(powerline) - 42ns
[TRACE] 09:10:03.142 text.go:Render( {{if .Root}}#{{else}}${{end}}) - 7.625µs
[TRACE] 09:10:03.142 text.go:Render(plain) - 83ns
[TRACE] 09:10:03.142 terminal.go:Shell() - 42ns
[TRACE] 09:10:03.142 terminal.go:Flags() - 41ns
[TRACE] 09:10:03.142 terminal.go:Flags() - 42ns
[TRACE] 09:10:03.142 terminal.go:GOOS() - 42ns
[DEBUG] 09:10:03.142 segment.go:Execute:103 → segment: Time
[TRACE] 09:10:03.142 terminal.go:GOOS() - 0s
[DEBUG] 09:10:03.142 segment.go:Execute:103 → segment: Terraform
[TRACE] 09:10:03.142 terminal.go:Flags() - 42ns
[DEBUG] 09:10:03.142 terraform.go:Enabled:35 → fetch_version: false
[TRACE] 09:10:03.143 terminal.go:GOOS() - 41ns
[DEBUG] 09:10:03.143 segment.go:Execute:103 → segment: Status
[TRACE] 09:10:03.143 terminal.go:StatusCodes() - 125ns
[DEBUG] 09:10:03.143 time.go:Enabled:27 → 15:04 (Mon)
[DEBUG] 09:10:03.143 status.go:formatStatus:45 → {{ .Code }}
[TRACE] 09:10:03.143 terminal.go:TemplateCache() - 42ns
[TRACE] 09:10:03.143 terminal.go:GOOS() - 42ns
[TRACE] 09:10:03.143 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.143 terminal.go:Flags() - 0s
[DEBUG] 09:10:03.143 segment.go:Execute:103 → segment: Kubectl
[DEBUG] 09:10:03.143 segment.go:Execute:103 → segment: Executiontime
[DEBUG] 09:10:03.143 kubectl.go:Enabled:45 → parse_kubeconfig: true
[DEBUG] 09:10:03.143 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.143 terminal.go:GOOS() - 0s
[TRACE] 09:10:03.143 terminal.go:Getenv(KUBECONFIG) - 1.583µs
[DEBUG] 09:10:03.143 executiontime.go:Enabled:55 → always_enabled: true
[DEBUG] 09:10:03.143 segment.go:Execute:103 → segment: Spotify
[TRACE] 09:10:03.143 terminal.go:ExecutionTime() - 42ns
[DEBUG] 09:10:03.143 executiontime.go:Enabled:57 → threshold: 500.000000
[DEBUG] 09:10:03.143 executiontime.go:Enabled:61 → austin
[TRACE] 09:10:03.143 terminal.go:TemplateCache() - 42ns
[DEBUG] 09:10:03.143 terminal.go:CommandPath:395 → /opt/homebrew/bin/terraform
[TRACE] 09:10:03.143 terminal.go:CommandPath(terraform) - 29.542µs
[TRACE] 09:10:03.143 text.go:Render({{ .Code }}) - 22.5µs
[TRACE] 09:10:03.143 terminal.go:HasCommand(terraform) - 31.041µs
[DEBUG] 09:10:03.143 status.go:Enabled:37 → always_enabled: true
[TRACE] 09:10:03.143 terminal.go:Flags() - 42ns
[TRACE] 09:10:03.143 terminal.go:TemplateCache() - 0s
[DEBUG] 09:10:03.143 terminal.go:HasFolder:272 → false
[TRACE] 09:10:03.143 terminal.go:GOOS() - 42ns
[TRACE] 09:10:03.143 terminal.go:HasFolder(/Users/the-mentor/.terraform) - 3.083µs
[DEBUG] 09:10:03.143 segment.go:Execute:103 → segment: Battery
[DEBUG] 09:10:03.143 terminal_unix.go:IsWsl:35 → false
[TRACE] 09:10:03.143 terminal_unix.go:IsWsl() - 3.625µs
[DEBUG] 09:10:03.143 terminal.go:FileContent:302 ↓
    apiVersion: v1
    kind: Config
    clusters:
      - name: rancher-desktop
        cluster:
          server: https://127.0.0.1:6443
          certificate-authority-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJkekNDQVIyZ0F3SUJBZ0lCQURBS0JnZ3Foa2pPUFFRREFqQWpNU0V3SHdZRFZRUUREQmhyTTNNdGMyVnkKZG1WeUxXTmhRREUzTWpVNU5UUTROVFF3SGhjTk1qUXdPVEV3TURjMU5ERTBXaGNOTXpRd09UQTRNRGMxTkRFMApXakFqTVNFd0h3WURWUVFEREJock0zTXRjMlZ5ZG1WeUxXTmhRREUzTWpVNU5UUTROVFF3V1RBVEJnY3Foa2pPClBRSUJCZ2dxaGtqT1BRTUJCd05DQUFUZVUyT3hvME8yZ0pua2NBRHFQaG14ZlVYL0xUSUtlRllNSjd1aS9YTWcKaUhBZVQ3Y2tOVkhWOWxQTEovWkJmRWlkbHBWeVRUUjk3d2lqb2VubzRUZUFvMEl3UURBT0JnTlZIUThCQWY4RQpCQU1DQXFRd0R3WURWUjBUQVFIL0JBVXdBd0VCL3pBZEJnTlZIUTRFRmdRVXZFWDVuZFdYSUxtdkdCUXVSQU1YCkdGV2dVcVF3Q2dZSUtvWkl6ajBFQXdJRFNBQXdSUUloQUtITldpZkFrbnljUHdXZzVlT1NqVDJHUEdmaSs1MDYKcE8vckFBVlpGQm9FQWlCK2RSRFA2K3JiOVNyOVlJMG52QWZKdDFMRVVNU28zLzR2SlNxY2dMcTNWUT09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0K
          insecure-skip-tls-verify: false
    users:
      - name: rancher-desktop
        user:
          client-certificate-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJrRENDQVRlZ0F3SUJBZ0lJRTNYUEpnYmpsM0l3Q2dZSUtvWkl6ajBFQXdJd0l6RWhNQjhHQTFVRUF3d1kKYXpOekxXTnNhV1Z1ZEMxallVQXhOekkxT1RVME9EVTBNQjRYRFRJME1Ea3hNREEzTlRReE5Gb1hEVEkxTURreApNREEzTlRReE5Gb3dNREVYTUJVR0ExVUVDaE1PYzNsemRHVnRPbTFoYzNSbGNuTXhGVEFUQmdOVkJBTVRESE41CmMzUmxiVHBoWkcxcGJqQlpNQk1HQnlxR1NNNDlBZ0VHQ0NxR1NNNDlBd0VIQTBJQUJBaGlPTThwbXh0cGpuRnEKbkIzUWRtRFcrZDV1eWJTRXAxek1VdnVSTkRYYjAyMHJBZkF1RElQR2gvUG9WcWdzZytUejVjYlhvZ3dZbC9nOQpIbHJDU1MralNEQkdNQTRHQTFVZER3RUIvd1FFQXdJRm9EQVRCZ05WSFNVRUREQUtCZ2dyQmdFRkJRY0RBakFmCkJnTlZIU01FR0RBV2dCU2RqRkt1Y0g2bjNPWElacldvQWU0blNScm1PREFLQmdncWhrak9QUVFEQWdOSEFEQkUKQWlBc21iaWh3ejB4V0l5RzV4TExKbmh2TVVkMGhvcVR6akVoblFTV2pkZEh6UUlnQm5EbmZhTnFQMzQzMGU2eQpzYkZ4enVlWGhya3JmQUlyRndTeDJTQmRsR3c9Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0KLS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUJkakNDQVIyZ0F3SUJBZ0lCQURBS0JnZ3Foa2pPUFFRREFqQWpNU0V3SHdZRFZRUUREQmhyTTNNdFkyeHAKWlc1MExXTmhRREUzTWpVNU5UUTROVFF3SGhjTk1qUXdPVEV3TURjMU5ERTBXaGNOTXpRd09UQTRNRGMxTkRFMApXakFqTVNFd0h3WURWUVFEREJock0zTXRZMnhwWlc1MExXTmhRREUzTWpVNU5UUTROVFF3V1RBVEJnY3Foa2pPClBRSUJCZ2dxaGtqT1BRTUJCd05DQUFRSlRiRWlSMW9uVC90dEY2TG5NeVIrUktyWEV5MkRDQmw5Y05vbmJJVXEKbnZBUDQ5aDhZMEhYYzd2VmVQWGRkcjNZTmJOaHpXZjkxVFpSSGFINzhSMi9vMEl3UURBT0JnTlZIUThCQWY4RQpCQU1DQXFRd0R3WURWUjBUQVFIL0JBVXdBd0VCL3pBZEJnTlZIUTRFRmdRVW5ZeFNybkIrcDl6bHlHYTFxQUh1Ckowa2E1amd3Q2dZSUtvWkl6ajBFQXdJRFJ3QXdSQUlnS2FJampXdEtsWTREZkVadHBpS0dUUGJwQjlMVk5JelQKdzc0bWgydkNQSUVDSUdNd1llRVowTmhUWHdhdEhFNTdxdnBLcnZST0g0MlR1dEJ1YkVjRVZ3L3UKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo=
          client-key-data: LS0tLS1CRUdJTiBFQyBQUklWQVRFIEtFWS0tLS0tCk1IY0NBUUVFSUl3dFhDcUxScWR4U0VtRWxZSHFQblF6WG9xOXphK1hqUmRyNGladitWVENvQW9HQ0NxR1NNNDkKQXdFSG9VUURRZ0FFQ0dJNHp5bWJHMm1PY1dxY0hkQjJZTmI1M203SnRJU25YTXhTKzVFME5kdlRiU3NCOEM0TQpnOGFIOCtoV3FDeUQ1UFBseHRlaURCaVgrRDBlV3NKSkx3PT0KLS0tLS1FTkQgRUMgUFJJVkFURSBLRVktLS0tLQo=
    contexts:
      - name: rancher-desktop
        context:
          cluster: rancher-desktop
          name: rancher-desktop
          user: rancher-desktop
    preferences: {}
    current-context: rancher-desktop

[TRACE] 09:10:03.143 terminal.go:FileContent(/Users/the-mentor/.kube/config) - 158.75µs
[TRACE] 09:10:03.143 terminal.go:TemplateCache() - 83ns
[DEBUG] 09:10:03.143 kubectl.go:SetContextAlias:156 → context_aliases: map[arn:aws:eks:ap-southeast-2:106376712998:cluster/eks01-au1:eks01-au1 arn:aws:eks:ca-central-1:106376712998:cluster/eks01-ca1:eks01-ca1 arn:aws:eks:eu-central-1:106376712998:cluster/eks01-eu2:eks01-eu2 arn:aws:eks:us-east-1:106376712998:cluster/eks01-us1:eks01-us1 arn:aws:eks:us-east-1:397966225531:cluster/eks01-dev:eks01-dev arn:aws:eks:us-east-1:397966225531:cluster/eks01-do1:eks01-do1 arn:aws:eks:us-east-1:397966225531:cluster/eks01-stg:eks01-stg]
[TRACE] 09:10:03.144 terminal.go:TemplateCache() - 42ns
[TRACE] 09:10:03.144 text.go:Render( {{ if gt .Code 0 }}󰀰{{ else }}{{ end }} ) - 998.041µs
[TRACE] 09:10:03.144 text.go:Render(plain) - 125ns
[TRACE] 09:10:03.144 text.go:Render({{ .FormattedMs }} ) - 38.333µs
[DEBUG] 09:10:03.144 terminal.go:HasFilesInDir:243 → false
[TRACE] 09:10:03.144 terminal.go:HasFilesInDir(.tf) - 1.110667ms
[TRACE] 09:10:03.144 text.go:Render(plain) - 334ns
[DEBUG] 09:10:03.144 terminal.go:HasFilesInDir:243 → false
[TRACE] 09:10:03.144 terminal.go:HasFilesInDir(.tfplan) - 3.75µs
[DEBUG] 09:10:03.144 terminal.go:HasFilesInDir:243 → false
[TRACE] 09:10:03.144 terminal.go:HasFilesInDir(.tfstate) - 2.958µs
[TRACE] 09:10:03.144 text.go:Render({{ .Context }}{{ if .Namespace }} :: {{ .Namespace }}{{ end }}) - 95.584µs
[TRACE] 09:10:03.144 text.go:Render(powerline) - 83ns
[TRACE] 09:10:03.144 text.go:Render(powerline) - 41ns
[DEBUG] 09:10:03.155 terminal.go:RunCommand:371 ↓
    Now drawing from 'AC Power'
     -InternalBattery-0 (id=73334883)	85%; AC attached; not charging present: true
[TRACE] 09:10:03.155 terminal.go:RunCommand(pmset -g batt) - 12.802708ms
[TRACE] 09:10:03.155 terminal_darwin.go:BatteryState() - 12.861917ms
[DEBUG] 09:10:03.155 battery.go:Enabled:53 → NO DATA
[TRACE] 09:10:03.155 terminal.go:TemplateCache() - 83ns
[DEBUG] 09:10:03.174 terminal.go:RunCommand:371 → true
[TRACE] 09:10:03.174 terminal.go:RunCommand(osascript -e application "Spotify" is running) - 31.686333ms
[DEBUG] 09:10:03.284 terminal.go:RunCommand:371 → paused
[TRACE] 09:10:03.284 terminal.go:RunCommand(osascript -e tell application "Spotify" to player state as string) - 110.22125ms
[DEBUG] 09:10:03.402 terminal.go:RunCommand:371 → Social Distortion
[TRACE] 09:10:03.402 terminal.go:RunCommand(osascript -e tell application "Spotify" to artist of current track as string) - 117.174708ms
[DEBUG] 09:10:03.515 terminal.go:RunCommand:371 → Ball and Chain
[TRACE] 09:10:03.515 terminal.go:RunCommand(osascript -e tell application "Spotify" to name of current track as string) - 113.668375ms
[DEBUG] 09:10:03.515 spotify.go:resolveIcon:43 → 󰏤
[TRACE] 09:10:03.515 terminal.go:TemplateCache() - 125ns
[TRACE] 09:10:03.515 terminal.go:TemplateCache() - 0s
[TRACE] 09:10:03.516 text.go:Render( {{ .Icon }}{{ if ne .Status "stopped" }}{{ .Artist }} - {{ .Track }}{{ end }} ) - 195.542µs
[TRACE] 09:10:03.516 text.go:Render(powerline) - 125ns
[TRACE] 09:10:03.516 text.go:Render( {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }}{{ .Error }}% ) - 14.5µs
[TRACE] 09:10:03.516 text.go:Render({{if eq "Charging" .State.String}}#40c4ff{{end}}) - 33.458µs
[TRACE] 09:10:03.516 text.go:Render({{if eq "Discharging" .State.String}}#ff5722{{end}}) - 9.458µs
[TRACE] 09:10:03.516 text.go:Render({{if eq "Full" .State.String}}#4caf50{{end}}) - 5.916µs
[TRACE] 09:10:03.516 text.go:Render(powerline) - 42ns
[TRACE] 09:10:03.516 text.go:Render({{if eq "Charging" .State.String}}#40c4ff{{end}}) - 5.917µs
[TRACE] 09:10:03.516 text.go:Render({{if eq "Discharging" .State.String}}#ff5722{{end}}) - 6.667µs
[TRACE] 09:10:03.516 text.go:Render({{if eq "Full" .State.String}}#4caf50{{end}}) - 7.083µs
[TRACE] 09:10:03.516 text.go:Render( {{ .CurrentDate | date .Format }} ) - 8.5µs
[TRACE] 09:10:03.516 text.go:Render(diamond) - 41ns
[TRACE] 09:10:03.516 text.go:Render({{if eq "Charging" .State.String}}#40c4ff{{end}}) - 8.292µs
[TRACE] 09:10:03.516 text.go:Render({{if eq "Discharging" .State.String}}#ff5722{{end}}) - 7.833µs
[TRACE] 09:10:03.516 text.go:Render({{if eq "Full" .State.String}}#4caf50{{end}}) - 5.25µs
[TRACE] 09:10:03.516 terminal.go:Shell() - 125ns
[TRACE] 09:10:03.516 terminal.go:Flags() - 42ns
[TRACE] 09:10:03.516 terminal.go:Shell() - 0s
[DEBUG] 09:10:03.522 terminal_unix.go:TerminalWidth:88 → terminal width: 189
[TRACE] 09:10:03.522 terminal_unix.go:TerminalWidth() - 6.384208ms
[DEBUG] 09:10:03.522 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.522 terminal.go:Getenv(LOCALAPPDATA) - 2.958µs
[DEBUG] 09:10:03.522 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.522 terminal.go:Getenv(OMP_CACHE_DIR) - 1.125µs
[DEBUG] 09:10:03.522 terminal.go:Getenv:161 → NO DATA
[TRACE] 09:10:03.522 terminal.go:Getenv(XDG_CACHE_HOME) - 1.125µs
[TRACE] 09:10:03.522 terminal.go:CachePath() - 24.5µs
[TRACE] 09:10:03.522 terminal.go:Flags() - 125nså
```

--- a/src/cache/duration.go
+++ b/src/cache/duration.go
@@ -9,6 +9,8 @@ type Duration string
 const (
 	INFINITE = Duration("infinite")
 	ONEWEEK  = Duration("168h")
+	ONEDAY   = Duration("24h")
+	TWOYEARS = Duration("17520h")
 )
 
 func (d Duration) Seconds() int {

--- a/src/cli/toggle.go
+++ b/src/cli/toggle.go
@@ -50,7 +50,7 @@ var toggleCmd = &cobra.Command{
 			newToggles = append(newToggles, segment)
 		}
 
-		env.Session().Set(cache.TOGGLECACHE, strings.Join(newToggles, ","), "1day")
+		env.Session().Set(cache.TOGGLECACHE, strings.Join(newToggles, ","), cache.ONEDAY)
 	},
 }
 

--- a/src/font/fonts.go
+++ b/src/font/fonts.go
@@ -78,7 +78,7 @@ func setCachedFontData(assets []*Asset) {
 		return
 	}
 
-	environment.Cache().Set(cache.FONTLISTCACHE, string(data), "1day")
+	environment.Cache().Set(cache.FONTLISTCACHE, string(data), cache.ONEDAY)
 }
 
 func CascadiaCode() ([]*Asset, error) {

--- a/src/runtime/http/oauth.go
+++ b/src/runtime/http/oauth.go
@@ -91,7 +91,7 @@ func (o *OAuthRequest) refreshToken(refreshToken string) (string, error) {
 
 	// add tokens to cache
 	o.Env.Cache().Set(o.AccessTokenKey, tokens.AccessToken, cache.ToDuration(tokens.ExpiresIn))
-	o.Env.Cache().Set(o.RefreshTokenKey, tokens.RefreshToken, "2years")
+	o.Env.Cache().Set(o.RefreshTokenKey, tokens.RefreshToken, cache.TWOYEARS)
 	return tokens.AccessToken, nil
 }
 

--- a/src/runtime/terminal.go
+++ b/src/runtime/terminal.go
@@ -591,7 +591,7 @@ func (term *Terminal) saveTemplateCache() {
 
 	templateCache, err := json.Marshal(tmplCache)
 	if err == nil {
-		term.sessionCache.Set(cache.TEMPLATECACHE, string(templateCache), "1day")
+		term.sessionCache.Set(cache.TEMPLATECACHE, string(templateCache), cache.ONEDAY)
 	}
 }
 
@@ -757,7 +757,7 @@ func (term *Terminal) setPromptCount() {
 	// Only update the count if we're generating a primary prompt.
 	if term.CmdFlags.Primary {
 		count++
-		term.Session().Set(cache.PROMPTCOUNTCACHE, strconv.Itoa(count), "1day")
+		term.Session().Set(cache.PROMPTCOUNTCACHE, strconv.Itoa(count), cache.ONEDAY)
 	}
 
 	term.CmdFlags.PromptCount = count

--- a/src/upgrade/notice.go
+++ b/src/upgrade/notice.go
@@ -6,6 +6,7 @@ import (
 	"time"
 
 	"github.com/jandedobbeleer/oh-my-posh/src/build"
+	"github.com/jandedobbeleer/oh-my-posh/src/cache"
 	"github.com/jandedobbeleer/oh-my-posh/src/runtime"
 	"github.com/jandedobbeleer/oh-my-posh/src/runtime/http"
 )
@@ -83,7 +84,7 @@ func Notice(env runtime.Environment, force bool) (string, bool) {
 		return "", false
 	}
 
-	env.Cache().Set(CACHEKEY, latest, "1week")
+	env.Cache().Set(CACHEKEY, latest, cache.ONEWEEK)
 
 	version := fmt.Sprintf("v%s", build.Version)
 	if latest == version {

